docs/System-Description-Format.md

```diff
@@ -78,6 +86,44 @@ A simplified example of the manifest can look like this:
 }
 ```
 
+## Data Structures
+
+The machinery data model supports two data structures: Arrays and Objects.
+
+Objects are just plain JSON objects , e.g.
+
+```
+  {
+    "modified": "2014-08-22T14:34:59Z",
+    "hostname": "host.example.org"
+  }
+```
+
+Arrays on the other hand are not just a collection of elements (of an arbitrary type) but can also
+have attributes defined on the list itself. This can be used to specify whether a list of
+packages contains RPM or DEB packages, for example.
+
+The notation for these arrays is a JSON object containing two elements, an attribute JSON object
+and an array containing the elements:
+
+```
+  {
+    "_attributes": {
+      "package_system": "rpm"
+    },
+    "_elements": [
+      {
+        "name": "acl",
+        "version": "2.2.52",
+        "release": "6.4"
+      }
+    ]
+  }
+```
+
+Array elements and Object values can be either primitive values or Arrays or Objects again. Plain
+JSON arrays are not allowed.
```

Comment:
What does "Plain JSON arrays are not allowed" mean? The content of the `_elements` attribute is a plain JSON array, isn't it? What happens if I use e.g. `"version": ["1.0", 2.0"]`?

This whole section is written from an implementation point of view. It probably would fit better into the data model documentation. For the format documentation users have to understand how to read the data, not how to construct new schemas.